File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None
    task_type: str


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: str | None = None


class ProjectResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    owner_id: uuid.UUID | None
    task_type: str
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/schemas/project.ts
import { z } from "zod";

export const ProjectCreateSchema = z.object({
  name: z.string().min(1, "Name is required"),
  description: z.string().optional(),
  task_type: z.enum(["pairwise_preference", "relevance_rating", "classification", "extraction_qa", "freeform_critique"]),
});

export const ProjectResponseSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  description: z.string().nullable(),
  owner_id: z.string().uuid().nullable(),
  task_type: z.string(),
  status: z.enum(["draft", "active", "paused", "completed"]),
  created_at: z.string(),
  updated_at: z.string(),
});

export type ProjectCreate = z.infer<typeof ProjectCreateSchema>;
export type ProjectResponse = z.infer<typeof ProjectResponseSchema>;
